pkg/cosign/tuf/client.go

```diff
@@ -235,66 +233,62 @@ func GetRootStatus(ctx context.Context) (*RootStatus, error) {
 //       targets in a targets/ subfolder.
 //   * forceUpdate: indicates checking the remote for an update, even when the local
 //       timestamp.json is up to date.
-func initializeTUF(ctx context.Context, mirror string, root []byte, embedded fs.FS, forceUpdate bool) (*TUF, error) {
-	singletonTUFMu.Lock()
-	defer singletonTUFMu.Unlock()
-	if singletonTUF != nil {
-		return singletonTUF, nil
-	}
+func initializeTUF(ctx context.Context, mirror string, root []byte, embedded fs.FS, forceUpdate bool) *TUF {
+	singletonTUFOnce.Do(func() {
+		t := &TUF{
+			mirror:   mirror,
+			embedded: embedded,
+		}
 
-	t := &TUF{
-		mirror:   mirror,
-		embedded: embedded,
-	}
+		t.targets = newFileImpl()
+		var err error
+		t.local, err = newLocalStore()
+		if err != nil {
+			panic(err)
```

Comment:
Darn, is this the consequence of sync.Once? If so, this isn't viable imo, we shouldn't add a panic into the client because if a server starts depending on it, it opens up an opportunity for query of deaths. Is there a way to return an error? If not, what you had before would be fine.

Reply:
+1 on not panicing.

Reply:
Could do a global variable - https://stackoverflow.com/questions/42069615/too-many-arguments-to-return-error

Just ran into this same issue in fulcioroots

Reply:
Using the global!